src/aioquic/quic/connection.py

```diff
@@ -398,6 +400,27 @@ def change_connection_id(self) -> None:
                 "Switching to CID %s (%d)", dump_cid(self._peer_cid), self._peer_cid_seq
             )
 
+    def reset_stream(
+        self,
+        stream_id: int,
+        error_code: int = QuicErrorCode.NO_ERROR,
+    ) -> None:
+        """
+        Reset a stream.
+
+        :param stream_id: The ID of the stream to reset.
+        :param error_code: (optional) An error code indicating why the connection is
+                           being reset. Default: `QuicErrorCode.NO_ERROR`
+        """
+        stream = self._streams[stream_id] = QuicStream(stream_id=stream_id)
+
+        if self._state not in END_STATES:
+            self._reset_event = events.StreamReset(
```

Comment:
Same comment as for `_reset_pending` : the error code is specific to a stream. I think it would be sufficient to store `_reset_error_code` on the stream.